torch/distributed/tensor/_random.py

```diff
@@ -159,11 +158,25 @@ class OffsetBasedRNGTracker(_RNGStateTracker):
     This subclass of ``_RNGStateTracker`` defines the default policy of how RNG states
     should be shared and synchronized among all ranks to respect the semantics of DTensor
     random operators.
+
+    note: _RNGStateTracker only supports cuda/cuda-like device
     """
 
-    def __init__(self, device_type: str = "cuda", run_state_sync: bool = True):
-        super().__init__(device_type)
-        rng_state = self._device_handle.get_rng_state().to(device_type)
+    def __init__(
+        self,
+        device_mesh: DeviceMesh,
+        run_state_sync: bool = True,
+    ):
+        super().__init__(_resolve_device(device_mesh=device_mesh))
+        assert self._device_handle is not None
+        # DTensor RNG tracker so far only supports CUDA/CUDA-like devices
+        if self._device.type != "cuda":
+            raise RuntimeError(
+                f"{self.__class__.__name__} instantiation requires the presence of "
+                f"CUDA/CUDA-like device. Got {self._device.type} instead."
```

Comment:
PR Lgtm overall. 
Remind me if we have a cpu-compatible rng tracker today?

Reply:
@wconstab unfortunately no, and it may not be very useful IMO, because the main usage of DTensor random ops is model init and `dropout()` in model training. Neither is expected to run on CPU. 

If we see a use case, it's also possible to achieve a CPU RNG tracker by either us or users. Users can also extend the base class to make a tracker for CPU. Let me know if we already have such a use case.